src/utils/validation.ts

```diff
@@ -0,0 +1,17 @@
+import { ZodError, ZodSchema } from "zod";
+import { InvalidArgumentError } from "./CommandError";
+
+export const validateField = <T>(schema: ZodSchema<T>, data: unknown): T => {
+  try {
+    return schema.parse(data);
+  } catch (error: unknown) {
+    if (error instanceof ZodError) {
+      // we don't want to spam the user with all the errors. Take the first one.
+      const message = error.issues[0].message;
```

Comment:
```suggestion
      // we don't want to spam the user with all the errors. Take the first one.
      const message = error.issues[0].message;
```

Why not? I prefer to see the bag of errors than filtering and rerunning the code to find the next problem

Reply:
It's something I was planning to change. If the user enters 10k events with a bad ID format, it will print 10k repeated errors.

Reply:
I would expect a single error `ZodError` and if that prints all the values under `issues` when logging, so be it!

In other word, don't hide `ZodError` and let them use `error.issues` however they want. At this layer of problem, it is almost a system error.